test/components/addressbook/addressbook.controller.js

```diff
@@ -69,13 +85,42 @@ describe('AddressbookController', function () {
       beforeEach(function () {
         storageServiceMock.get = sinon.stub().returns(null)
       })
-
       it('sets controller contacts to empty array on invalid return value', () => {
         ctrl.getContacts()
 
         expect(storageServiceMock.get.calledOnce).to.be.true
         expect(ctrl.contacts).to.deep.equal([])
       })
     })
+
+    context("set up addressbook contact modal", () => {
+      it('sets up address book modal', () => {
+        ctrl.addAddressbookContact()
+        sinon.assert.calledOnce(mdDialogShowStub)
+        expect($scope.addAddressbookContact).to.have.all.keys(['add', 'cancel'])
+        expect(typeof $scope.addAddressbookContact.add).to.equal('function')
+        expect(typeof $scope.addAddressbookContact.cancel).to.equal('function')
+      })
+    })
+
+    context("Adding contacts", () => {
+      beforeEach(function () {
+        ctrl.getContacts = sinon.stub().returns([])
+        ctrl.showToast = sinon.stub()
+        ctrl.isAddress = sinon.stub()
+      })
+      it('should fail to add due to empty name', () => {
+        let name = ''
+        let address = 'a'
+        ctrl.addAddressbookContact()
+        const retVal = $scope.addAddressbookContact.add(name, address)
+        sinon.assert.calledOnce(mdDialogShowStub)
+        sinon.assert.match(retVal, undefined)
+        sinon.assert.calledOnce(ctrl.getContacts)
+        sinon.assert.match(ctrl.trim(name), '')
+        sinon.assert.calledWith(ctrl.showToast, 'This Contact Name is not valid', '', true)
```

Comment:
Be careful when testing against explicit strings, if someone was to change the wording for messages in the app, this assert would fail (even though the functionality is left unchanged).